### DISCUSSION TITLE
Adding content items in a DataMigration?

### DISCUSSION THREAD

**Discussion by NickDarvey** [OrchardCMS/OrchardCore#8628] [Q&A] [answered]
If I understand correctly, you can express migrations of content definitions in two ways:
Code, like in [this Orchard Dojo example](https://github.com/Lombiq/Dojo-Course-3-Orchard-Core-Demo/blob/master/src/Modules/DojoCourse.Module/Migrations/PersonMigrations.cs), or
JSON, like in [The Blog Theme](https://github.com/OrchardCMS/OrchardCore/blob/a27f2de7b8889e1a7649c23b819ff372fbcb8aa1/src/OrchardCore.Themes/TheBlogTheme/Recipes/blog.recipe.json).

I can see the JSON recipes also add content items like:
https://github.com/OrchardCMS/OrchardCore/blob/a27f2de7b8889e1a7649c23b819ff372fbcb8aa1/src/OrchardCore.Themes/TheBlogTheme/Recipes/blog.recipe.json#L1113-L1147
but how do you do this in a DataMigration?

My use case is similar to the JSON example I included, creating the one (and likely only) of something with my module.

**Comment by ns8482e** [comment, marked as answer]
1. Create `your-recipe.json` Recipe file in your module 
2. Inject `IRecipeMigrator` service in your `Migration.cs`
3. Call `ExecuteAsync`  with `your-recipe.json` file as parameter  in migration method.

An example from Menu module.

https://github.com/OrchardCMS/OrchardCore/blob/bf9231ef76afef6e5b5434616d29c50a783651dd/src/OrchardCore.Modules/OrchardCore.Menu/Migrations.cs#L18
